app/controllers/concerns/foreman/controller/puppet/hosts_controller_extensions.rb

```diff
@@ -0,0 +1,164 @@
+module Foreman::Controller::Puppet::HostsControllerExtensions
+  extend ActiveSupport::Concern
+
+  PUPPETMASTER_ACTIONS=[ :externalNodes, :lookup ]
+  PUPPET_AJAX_REQUESTS=%w{hostgroup_or_environment_selected puppetclass_parameters}
+
+  PUPPET_MULTIPLE_ACTIONS = %w(select_multiple_environment update_multiple_environment
+                               multiple_puppetrun update_multiple_puppetrun
+                               select_multiple_puppet_proxy update_multiple_puppet_proxy
+                               select_multiple_puppet_ca_proxy update_multiple_puppet_ca_proxy)
```

Comment:
i wonder if we should take the opportunity to refactor all of the bulk actions to a dedicated controller too.